examples/file_upload/src/main.rs

```diff
@@ -38,33 +37,28 @@ impl Component for App {
 
     fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
         match msg {
-            Msg::Loaded(file_name, file_type, data) => {
-                self.files.push(FileDetails {
-                    data,
-                    file_type,
-                    name: file_name.clone(),
-                });
-                self.readers.remove(&file_name);
+            Msg::Loaded(file) => {
+                let name = file.name.clone();
+                self.files.push(file);
+                self.readers.remove(&name);
```

Comment:
If you flip these two lines, you should be able to avoid the clone above. It doesn't really matter though